pkg/helm/actions/install_chart.go

```diff
@@ -25,6 +25,10 @@ func InstallChart(ns, name, url string, vals map[string]interface{}, conf *actio
 		return nil, err
 	}
 
+	// Add chart URL as an annotation before installation
+	ch.Metadata.Annotations = make(map[string]string)
```

Comment:
Should I safety check that Metadata is not nil here?

Reply:
Not sure... I think since ch was loaded and there was no error I would expect Metadata to be there. However you brought up another possibility which is the charts do come with some annotation. In that case we should not create a new instance but re-use the exiting one there.

Reply:
I think you should check and use the structure if available.

Reply:
Yea we don't want to flush the existing annotations. Will check for existing Annotations here.

Reply:
Updated.